lua/nvim_lsp/elixirls.lua

```diff
@@ -0,0 +1,110 @@
+local configs = require 'nvim_lsp/configs'
+local util = require 'nvim_lsp/util'
+
+local server_name = "elixirls"
+local bin_name = "elixir-ls"
+local bin
+
+local function make_installer()
+    local P = util.path.join
+    local install_dir = P{util.base_install_dir, server_name}
+    local git_dir = P{install_dir, bin_name}
+
+    if vim.fn.has('mac') == 1 or vim.fn.has('unix') == 1 then
+        bin = P{git_dir, "release", "language_server.sh"}
+    elseif vim.fn.has('win32') == 1  or vim.fn.has('win64') == 1 then
+        bin = P{git_dir, "release", "language_server.bat"}
+    else
+        error("System is not supported, try to install manually.")
+        return
+    end
+
+    local X = {}
+    function X.install()
+        local install_info = X.info()
+        if install_info.is_installed then
+            print(server_name, "is already installed.")
+            return
+        end
+
+        if not (util.has_bins("elixir") and util.has_bins("erl")) then
+            error("Need elixir and erl to install this")
+            return
+        end
+
+        local script = [=[
+            set -e
+
+            # clone project
+            git clone https://github.com/elixir-lsp/elixir-ls
+            cd elixir-ls
+            
+            # fetch dependencies and compile
+            mix deps.get && mix compile
+
+            # install executable
+            mix elixir_ls.release -o release
+        ]=]
+        vim.fn.mkdir(install_info.install_dir, "p")
+        util.sh(script, install_info.install_dir)
+    end
+
+    function X.info()
+        return {
+            is_installed = util.path.exists(bin);
+            install_dir = install_dir;
+            cmd = { bin };
+        }
+    end
+
+    function X.configure(config)
+        local install_info = X.info()
+        if install_info.is_installed then
+            config.cmd = install_info.cmd
+        end
+    end
+    return X
+end
+
+local installer = make_installer()
+
+configs[server_name] = {
+  default_config = {
+    -- cmd = {bin};
```

Comment:
```suggestion
    cmd = { cmd_path };
```
https://github.com/neovim/nvim-lsp/pull/243/files#r427172387

Reply:
No, I think this should be "cmd". If users put bin at directory through global path, cmd is executable.

Reply:
I'm not entirely sure if this is correct, but wouldn't this just wrap the cmd string which holds the binary and not the complete path?

Reply:
Yes, AFAIK, elixr's ls formula was to run this script instead of the binary, so if I put something in the path, wouldn't it be this script?

Reply:
Yes, I just forgot about how global path works for a second...